projects/plugins/jetpack/modules/custom-css/custom-css.php

```diff
@@ -1,1795 +1,962 @@
 <?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
 
 use Automattic\Jetpack\Assets;
-use Automattic\Jetpack\Device_Detection\User_Agent_Info;
-use Automattic\Jetpack\Redirect;
 
 /**
- * Custom CSS class.
+ * Alternate Custom CSS source for 4.7 compat.
+ *
+ * @since 4.4.2
+ *
+ * @package automattic/jetpack
  */
-class Jetpack_Custom_CSS {
+
+/**
+ * Class Jetpack_Custom_CSS_Enhancements
+ */
+class Jetpack_Custom_CSS_Enhancements {
 
 	/**
-	 * Initialize the class.
+	 * Set up the actions and filters needed for our compatability layer on top of core's Custom CSS implementation.
 	 */
-	public static function init() {
-		add_action( 'switch_theme', array( __CLASS__, 'reset' ) );
-		add_action( 'wp_restore_post_revision', array( __CLASS__, 'restore_revision' ), 10, 2 );
-
-		// Save revisions for posts of type safecss.
-		add_action( 'load-revision.php', array( __CLASS__, 'add_revision_redirect' ) );
-
-		// Override the edit link, the default link causes a redirect loop
-		add_filter( 'get_edit_post_link', array( __CLASS__, 'revision_post_link' ), 10, 3 );
-
-		// Overwrite the content width global variable if one is set in the custom css
+	public static function add_hooks() {
+		add_action( 'init', array( __CLASS__, 'init' ) );
+		add_action( 'customize_controls_enqueue_scripts', array( __CLASS__, 'customize_controls_enqueue_scripts' ) );
+		add_action( 'customize_register', array( __CLASS__, 'customize_register' ) );
+		add_filter( 'map_meta_cap', array( __CLASS__, 'map_meta_cap' ), 20, 2 );
+		add_action( 'customize_preview_init', array( __CLASS__, 'customize_preview_init' ) );
+		add_filter( '_wp_post_revision_fields', array( __CLASS__, 'wp_post_revision_fields' ), 10, 2 );
+		add_action( 'load-revision.php', array( __CLASS__, 'load_revision_php' ) );
+
+		add_action( 'wp_enqueue_scripts', array( __CLASS__, 'wp_enqueue_scripts' ) );
+
+		// Handle Sass/LESS.
+		add_filter( 'customize_value_custom_css', array( __CLASS__, 'customize_value_custom_css' ), 10, 2 );
+		add_filter( 'customize_update_custom_css_post_content_args', array( __CLASS__, 'customize_update_custom_css_post_content_args' ), 10, 3 );
+		add_filter( 'update_custom_css_data', array( __CLASS__, 'update_custom_css_data' ), 10, 2 );
+
+		// Stuff for stripping out the theme's default stylesheet...
+		add_filter( 'stylesheet_uri', array( __CLASS__, 'style_filter' ) );
+		add_filter( 'safecss_skip_stylesheet', array( __CLASS__, 'preview_skip_stylesheet' ) );
+
+		// Stuff for overriding content width...
+		add_action( 'customize_preview_init', array( __CLASS__, 'preview_content_width' ) );
+		add_filter( 'jetpack_content_width', array( __CLASS__, 'jetpack_content_width' ) );
+		add_filter( 'editor_max_image_size', array( __CLASS__, 'editor_max_image_size' ), 10, 3 );
 		add_action( 'template_redirect', array( __CLASS__, 'set_content_width' ) );
 		add_action( 'admin_init', array( __CLASS__, 'set_content_width' ) );
```

Comment:
Looks like the rebase managed to lose one closing brace in the file
```suggestion
		add_action( 'admin_init', array( __CLASS__, 'set_content_width' ) );
	}
```